libraries/AP_NavEKF3/AP_NavEKF3_Control.cpp

```diff
@@ -507,10 +509,11 @@ bool NavEKF3_core::assume_zero_sideslip(void) const
 // set the LLH location of the filters NED origin
 bool NavEKF3_core::setOriginLLH(const Location &loc)
 {
-    if ((PV_AidingMode == AID_ABSOLUTE) && (frontend->_fusionModeGPS != 3)) {
+    if ((PV_AidingMode == AID_ABSOLUTE) && (frontend->_sources.getPosXYSource() == AP_NavEKF_Source::SourceXY::GPS)) {
```

Comment:
a != becoming an == ??

Reply:
Yes, I think that in this case at least it should change from a "!=" to an "=" because in master we have 4 fusion modes  ([see here](https://github.com/ArduPilot/ardupilot/blob/master/libraries/AP_NavEKF3/AP_NavEKF3.cpp#L135)):

   0:GPS 3D Vel and 2D Pos
   1:GPS 2D vel and 2D pos
   2:GPS 2D pos
   3:No GPS

So the old code was really saying, "Not Non-GPS" which is equivalent to "GPS".  In the new code we are more specific about "Non-GPS" because we expect the user to specify exactly which source they are using:

   0: None
   3: GPS
   4: Beacon
   5: External Nav